Sources/NIOCore/ByteBuffer-core.swift

```diff
@@ -1119,3 +1117,100 @@ extension ByteBuffer {
+    ///         including empty bytes in the end of the allocated space. Defaults to true.
+    ///     - limit: The maximum amount of bytes presented in the dump.
+    func hexDumpShort(offset: Int = 0, readableOnly: Bool = true, limit: Int) -> String {
+        let lastIndex = readableOnly ? self.writerIndex : self.capacity
+        let length = lastIndex - offset
+
+        // If the length of the resulting hex dump is smaller than the maximum allowed dump length,
+        // return the full dump. Otherwise, dump the first and last pieces of the buffer, then concatenate them.
+        if length <= limit {
+            return self.hexDumpShort(offset: offset, readableOnly: readableOnly)
+        } else {
+            let clipLength = limit / 2
+            let startHex = self._storage.dumpBytes(slice: self._storage.fullSlice,
+                                                   offset: offset,
+                                                   length: offset + clipLength)
+
+            let endHex = self._storage.dumpBytes(slice: self._storage.fullSlice,
+                                                 offset: lastIndex - clipLength,
+                                                 length: clipLength)
+
+            return startHex + " ... " + endHex
+        }
+    }
+
+    /// Returns a `String` of hexadecimal digits of bytes in the Buffer,
+    /// with formatting compatible with output of `hexdump -C`.
+    func hexDumpLong(offset: Int = 0, readableOnly: Bool = true) -> String {
+        var result = ""
+        let lastIndex = (readableOnly ? self.writerIndex : self.capacity)
+        let length = lastIndex - offset
+
+        // Grab the pointer to the whole buffer storage. We'll use offset and length when we index into it.
+        let bytes = UnsafeRawBufferPointer(start: self._storage.bytes, count: Int(self.capacity))
+
+        // hexdump -C dumps into lines of upto 16 bytes each
+        // with a last line containing only the offset column.
+        for line in 0 ... length/16 {
+            // Line offset column
+            result += "\(String(byte: line * 16, padding: 8))  "
+
+            // The amount of bytes remaining in this dump
+            let remainingBytes = length - line * 16
+
+            // The amount of bytes in this dump line.
+            let lineLength = min(16, remainingBytes)
+
+            // The index range in the buffer storage for this line.
+            let range = (offset + line*16)..<(offset + line*16 + lineLength)
+
+            // Grab just the bytes for this particular line
+            let lineBytes = bytes[range]
+
+            // Make a hexdump of a single line of up to 16 bytes
+            // Pad it with empty space if there were not enough bytes in the buffer dump range
+            // And insert the separator space in the middle of the line
+            var lineDump = bytes[range].map { String(byte: $0, padding: 2) }.joined(separator: " ")
```

Comment:
This is effectively `hexDumpShort` -- if you use the `slice` as suggested above you can make use of it here too.